File: sourceFile/Java/283_MoveZeros.java
/*
 * @Author: AlanGolphi
 * @Date: 2020-07-18 11:12:16
 * @LastEditTime: 2020-07-20 16:40:41
 */ 

class Solution {
    public void MoveZeros (int[] nums) {
        int j = 0;
        for (int i = 0;i < nums.length; ++i) {
            if (nums[i] != 0) {
                nums[j] = nums[i];
                if (i != j) {
                    nums[i] = 0;
                }
                j ++;
            }
        }
    }
}
//双指针法，j初始位置在 0 ，i从 0 开始遍历
//如果 i 的位置不是 0 ，就把 第 i 个数赋值给 j 的位置，j++
//如果 i 和 j 不在同一个位置，执行完上面的操作后就把第 i 个数置零

File: sourceFile/Js/283_MoveZeros.js
/*
 * @Author: AlanGolphi
 * @Date: 2021-03-27 18:09:31
 * @LastEditTime: 2021-03-27 18:16:26
 */
function moveZeros(array) {
    let j = 0;
    let len = array.length;

    for (let i = 0; i < len; ++i) {
        if (array[i] != 0) {
            array[j] = array[i];
            if (i != j) {
                array[i] = 0;
            }
            j++;
        }
    }
    return array;
}

let test = [0,1,0,3,6,0,8,0,0];
console.log(test);
console.log(moveZeros(test));